include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -573,6 +652,20 @@ class __future : private std::tuple<_Args...>
     }
 };
 
+// Only use USM host allocations on Intel GPUs. Other devices show significant slowdowns.
+inline bool
+__use_USM_host_allocations(sycl::queue __queue)
+{
+    auto __device = __queue.get_device();
+    if (!__device.is_gpu())
+        return false;
+    if (!__device.has(sycl::aspect::usm_host_allocations))
+        return false;
+    if (__device.get_info<sycl::info::device::vendor_id>() != 32902)
+        return false;
```

Comment:
Filtering out some code paths based on vendor id instead of HW capabilities might cause bad perception. What alternative checks might be used here?

Reply:
I think the key characteristic is the backend. L0 provides a low-overhead implementation of USM host memory whereas other backends use slower pinned host memory (mlock on Linux).